clients/gateway.go

```diff
@@ -161,27 +167,32 @@ type ExecutionResources struct {
 }
 
 type TransactionReceipt struct {
-	ActualFee          *felt.Felt          `json:"actual_fee"`
-	Events             []*Event            `json:"events"`
-	ExecutionResources *ExecutionResources `json:"execution_resources"`
-	L1ToL2Message      *L1ToL2Message      `json:"l1_to_l2_consumed_message"`
-	L2ToL1Message      *[]L2ToL1Message    `json:"l2_to_l1_messages"`
-	TransactionHash    *felt.Felt          `json:"transaction_hash"`
-	TransactionIndex   *big.Int            `json:"transaction_index"`
+	ActualFee                *felt.Felt                `json:"actual_fee"`
+	Events                   []*Event                  `json:"events"`
+	ExecutionResources       *ExecutionResources       `json:"execution_resources"`
+	L1ToL2Message            *L1ToL2Message            `json:"l1_to_l2_consumed_message"`
+	L2ToL1Message            []*L2ToL1Message          `json:"l2_to_l1_messages"`
+	TransactionHash          *felt.Felt                `json:"transaction_hash"`
+	TransactionIndex         *big.Int                  `json:"transaction_index"`
+	Status                   string                    `json:"status"`
```

Comment:
This and all the fields after it should go in a separate struct called `TransactionReceiptStatus`. Similar to how `Transaction` and `TransactionStatus` works.